### PR TITLE
hw 3 first commit

### PR DESCRIPTION
**Pull request by kalimullina**

### CODE REVIEW

File: src/main/java/com/epam/tc/hw3/IndexPage.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    public String getServiceItemOnHeaderSection() {
+        return serviceItemOnHeaderSection.getText();
+    }
+
+    public String getMetalsColorsItemOnHeaderSection() {
+        return metalsColorsItemOnHeaderSection.getText();
+    }
+
+    public void checkFourImages() {
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw3/IndexPage.java]: this method does literally nothing.
here should be assertions

File: src/main/java/com/epam/tc/hw3/IndexPage.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        } else if (number == 3) {
+            return thirdTextUnderIcons.getText();
+        } else if (number == 4) {
+            return fourthTextUnderIcons.getText();
+        } else {
+            return null;
+        }
+    }
+
+    public void iframeIsExisted() {
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw3/IndexPage.java]: this method does nothing.
rework it

File: src/main/java/com/epam/tc/hw3/IndexPage.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        iframeWithFrameButton.isDisplayed();
+    }
+
+    public WebElement returnIframe() {
+        return iframeWithFrameButton;
+    }
+
+    public void checkIframeButton() {
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw3/IndexPage.java]: this method does nothing.
rework it

File: src/main/java/com/epam/tc/hw3/IndexPage.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+    public void checkFourImages() {
+        firstImage.isDisplayed();
+        secondImage.isDisplayed();
+        thirdImage.isDisplayed();
+        fourthImage.isDisplayed();
+    }
+
+    public String getTextUnderIcons(int number) {
+        if (number == 1) {
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw3/IndexPage.java]: rewrite it with switch case

File: src/main/java/com/epam/tc/hw3/DifferentElementsPage.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    @FindBy(tagName = "option")
+    private List<WebElement> colorsInDropdown;
+
+    @FindBy(css = "ul[class='panel-body-list logs']")
+    private WebElement logRows;
+
+
+
+    public DifferentElementsPage selectWaterCheckbox() {
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw3/DifferentElementsPage.java:26]: one of the requirements of hw3 is using Void Page Object pattern in Page Object.
Fix all methods, which returning the page instance

File: src/main/java/com/epam/tc/hw3/DifferentElementsPage.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public DifferentElementsPage selectYellowInDropdown() {
+        dropdownWithColors.click();
+        colorsInDropdown.get(3).click();
+        return this;
+    }
+
+    public String getLogRows() {
+        String result = logRows.getText();
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw3/DifferentElementsPage.java]: you can just return logsRows.getText()

File: src/test/java/com/epam/tc/hw3/TestBase.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        webDriver = new ChromeDriver();
+        webDriver.manage().window().maximize();
+
+        homePage = PageFactory.initElements(webDriver, HomePage.class);
+        indexPage = PageFactory.initElements(webDriver, IndexPage.class);
+        differentElementsPage = PageFactory.initElements(webDriver, DifferentElementsPage.class);
+
+        property = new Properties();
+        file = new FileInputStream("src/test/resources/test.properties");
+        property.load(file);
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/TestBase.java:36]: where are the quit from FileInputStream?

File: src/main/java/com/epam/tc/hw3/IndexPage.java (hunk 1 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw3/IndexPage.java]: fixed

File: src/main/java/com/epam/tc/hw3/IndexPage.java (hunk 2 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw3/IndexPage.java]: fixed

File: src/main/java/com/epam/tc/hw3/IndexPage.java (hunk 3 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw3/IndexPage.java]: fixed

File: src/main/java/com/epam/tc/hw3/IndexPage.java (hunk 4 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw3/IndexPage.java]: fixed

File: src/main/java/com/epam/tc/hw3/DifferentElementsPage.java (hunk 6 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw3/DifferentElementsPage.java]: Sorry, very stupid

File: src/test/java/com/epam/tc/hw3/TestBase.java (hunk 7 above)
Comment (kalimullina) [reply src/test/java/com/epam/tc/hw3/TestBase.java:36]: fixed

File: src/main/java/com/epam/tc/hw3/IndexPage.java (hunk 1 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw3/IndexPage.java]: Can we have Assert in the page method?

File: src/main/java/com/epam/tc/hw3/IndexPage.java [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+            case 1:
+                return firstTextUnderIcons.getText();
+            case 2:
+                return secondTextUnderIcons.getText();
+            case 3:
+                return thirdTextUnderIcons.getText();
+            case 4:
+                return fourthTextUnderIcons.getText();
+            default:
+                return null;
```
Comment (ZeroShadows) [inline src/main/java/com/epam/tc/hw3/IndexPage.java]: lets not return null as default value, it would cause the NullPointerException.

**Review by ZeroShadows — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/tc/hw3/IndexPage.java (hunk 8 above)
Comment (kalimullina) [reply src/main/java/com/epam/tc/hw3/IndexPage.java]: fixed

**Review by ZeroShadows — APPROVED** [review]

**Pull request merged**